bg_atlasgen/validate_atlases.py

```diff
@@ -0,0 +1,195 @@
+
+    # minimum and maximum values of the annotation image scaled by the atlas resolution
+    z_min_scaled, z_max_scaled = z_min * resolution[0], z_max * resolution[0]
+    y_min_scaled, y_max_scaled = y_min * resolution[1], y_max * resolution[1]
+    x_min_scaled, x_max_scaled = x_min * resolution[2], x_max * resolution[2]
+
+    # z, y and x coordinates of the root mesh (extent of the whole object)
+    mesh_points = root_mesh.points
+    z_coords, y_coords, x_coords = (
+        mesh_points[:, 0],
+        mesh_points[:, 1],
+        mesh_points[:, 2],
+    )
+
+    # minimum and maximum coordinates of the root mesh
+    z_min_mesh, z_max_mesh = np.min(z_coords), np.max(z_coords)
+    y_min_mesh, y_max_mesh = np.min(y_coords), np.max(y_coords)
+    x_min_mesh, x_max_mesh = np.min(x_coords), np.max(x_coords)
+
+    # checking if root mesh and image are on the same scale
+    _assert_close(z_min_mesh, z_min_scaled, resolution[0])
+    _assert_close(z_max_mesh, z_max_scaled, resolution[0])
+    _assert_close(y_min_mesh, y_min_scaled, resolution[1])
+    _assert_close(y_max_mesh, y_max_scaled, resolution[1])
+    _assert_close(x_min_mesh, x_min_scaled, resolution[2])
+    _assert_close(x_max_mesh, x_max_scaled, resolution[2])
+
+    return True
+
+
+def open_for_visual_check():
+    pass
+
+
+def validate_checksum():
+    pass
+
+
+def check_additional_references():
+    # check additional references are different, but have same dimensions
+    pass
+
+
+def validate_atlas(atlas_name, version, all_validation_functions):
+    """Validates the latest version of a given atlas"""
+
+    print(atlas_name, version)
+    BrainGlobeAtlas(atlas_name)
+    updated = get_atlases_lastversions()[atlas_name]["updated"]
+    if not updated:
+        update_atlas(atlas_name)
+    Path(get_brainglobe_dir()) / f"{atlas_name}_v{version}"
+
+    validation_function_parameters = [
+        # validate_atlas_files(atlas_path: Path)
+        (Path(get_brainglobe_dir() / f"{atlas_name}_v{version}"),),
+        # validate_mesh_matches_image_extents(atlas: BrainGlobeAtlas)
+        (BrainGlobeAtlas(atlas_name),),
+        # open_for_visual_check()
+        (),
+        # validate_checksum()
+        (),
+        # check_additional_references()
+        (),
+        # validate_atlas(atlas_name, version)
+        (atlas_name, version),
+    ]
+
+    # list to store the errors of the failed validations
+    failed_validations = []
+    successful_validations = []
+
+    for i, validation_function in enumerate(all_validation_functions):
+        try:
+            validation_function(*validation_function_parameters[i])
+            successful_validations.append((atlas_name, validation_function))
+        except AssertionError as error:
+            failed_validations.append((atlas_name, validation_function, error))
+
+    return successful_validations, failed_validations
+
+
+def validate_mesh_structure_pairs(atlas_path: Path):
```

Comment:
We should at least make an issue to write tests for this function if we don't add them in this PR.